src/regex.ts

```diff
@@ -17,7 +17,7 @@ const protocol = `(https?:|ftps?:)\\/\\/`;
 const confirmedByProtocol = `(${protocol})\\S+`;
 const additionalSlashes = `(([\\/]?))+`;
 const fqdn = `(((${protocol})?(${domainWithTLD}|${ipv4}|(${protocol})(${ipv6}|${domainWithAnyTLD}))(?!@\\w)${port})|(${confirmedByProtocol}))`;
-const nonLatinMatches = `${fqdn}((((\\/(([${allowedInPath}]+(\\/[${allowedInPath}${nonLatinAlphabetRanges}]*)*))?)?)((\\?([${allowedInPath}\\/?]*))?)((\\#([${allowedInPath}\\/?]*))?))?\\b((([${allowedInPath}\\/${nonLatinAlphabetRanges}][a-zA-Z\\d\\-_~+=\\/${nonLatinAlphabetRanges}]+)?))+)`;
+const nonLatinMatches = `${fqdn}((((\\/(([${allowedInPath}]+(\\/[${allowedInPath}${nonLatinAlphabetRanges}]*)*))?)?)((\\?([${allowedInPath}\\/?]*))?)((\\#([${allowedInPath}\\/?]*))?))?\\b((([${allowedInPath}\\/${nonLatinAlphabetRanges}][a-zA-Z\\d\\-_~+=\\/${nonLatinAlphabetRanges}]+$)?))+)`;

```

Comment:
That's the only change I made. All other artifacts came from rebuilding. Left them as is to be consistent with the source.